templates/blocks/meta_fields.twig

```diff
@@ -1,19 +1,21 @@
 {% if ((post and (1 != fn('post_password_required', post))) or author is defined) %}
-	{% if (post.post_excerpt is defined and post.post_excerpt) %}
-		{% set meta_description = full_description %}
-	{% elseif (og_description is defined and og_description) %}
-		{% set meta_description = og_description %}
-	{% elseif (author.description is defined and author.description) %}
+  {% if (author.description is defined and author.description) %}
 		{% set meta_description = author.description|striptags %}
-	{% else %}
+  {% elseif (og_description is defined and og_description) %}
+		{% set meta_description = og_description %}
+  {% elseif (post.post_excerpt is defined and post.post_excerpt) %}
+		{% set meta_description = post.post_excerpt|striptags %}
+  {% else %}
 		{% set meta_description = site.description %}
 	{% endif %}
+
 	{% if (author.name is defined and author.name is not null) %}
 		{% set meta_twitter_creator = author.name %}
 	{% elseif (post.author is defined and post.author is not null) %}
 		{% set meta_twitter_creator = post.author.name %}
 	{% endif %}
-	<meta name="description" content="{{ meta_description|raw }}">
+
+	<meta name="description" content="{{ meta_description }}">
```

Comment:
Move the `|striptags` to this line.
We should always escape the string in the last step. The reason is:

1. Someone else may need to update the string before that step. If you escape the string at the beginning, others can not change it. 
2. Escape the string at last step can make sure all the strings are escaped.